code-studio/src/js/progress.js

```diff
@@ -160,12 +166,61 @@ progress.renderStageProgress = function (stageData, progressData, clientProgress
 };
 
 progress.renderCourseProgress = function (scriptData) {
-  var teacherCourse = $('#landingpage').hasClass('teacher-course');
+  var store = loadProgress(scriptData);
   var mountPoint = document.createElement('div');
+
+  $.ajax('/api/user_progress/' + scriptData.name).done(function (data) {
+    data = data || {};
+
+    // Show lesson plan links if teacher
+    if (data.isTeacher) {
+      $('.stage-lesson-plan-link').show();
+    }
+
+    // Merge progress from server (loaded via AJAX)
+    if (data.levels) {
+      store.dispatch({
+        type: 'MERGE_PROGRESS',
+        progress: _.mapValues(data.levels, level => {
+          return level.submitted ? 1000 : level.result;
+        })
```

Comment:
Could use `args => expression` arrow function here instead of `args => block`, i.e.

```javascript
progress: _.mapValues(data.levels, level => level.submitted ? 1000 : level.result)
```